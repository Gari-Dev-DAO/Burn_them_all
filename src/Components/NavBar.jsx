import { BsXLg } from "react-icons/bs";
import { TfiAlignJustify } from "react-icons/tfi";
import ConnetWalletButton from "./ConnetWalletButton"
import Button from "./Button";
import { useToggle } from "../hooks/useToggle";


const NavBar = () => {
  const [isToggle,makeToggle]=useToggle()
  return (
    <div className='navBar'>
       <div className="toggleBar"><Button onClick={makeToggle}>{!isToggle?<TfiAlignJustify size='32' color='white'/>:<BsXLg size={32} color='white'/>}</Button></div>
       <div  className={`nav ${isToggle ? "hide" : ""}`} data-testid="nav">
        <div className="navLinkBox">
         <a className="link">Go bonk Errs</a>
        </div>
        <div  className="navLinkBox navlinks" >
        <a href="https://www.coingecko.com/en/coins/bonk" target='_blank' className="link">Buy Bonk</a>
         <a href="https://web.telegram.org/k/#-813184380" target='_blank' className="link">Telegram</a>
        </div>
        <div className="navLinkBox">
        <ConnetWalletButton/>
        </div>
        </div>
    </div>
   
  )
}


export default NavBar