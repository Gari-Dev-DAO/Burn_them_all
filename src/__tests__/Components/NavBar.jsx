import { fireEvent, getByRole, render } from "@testing-library/react";
import NavBar from "../../Components/NavBar";



describe("NavBar", () => {
  it("renders component", () => {
    const { container } = render(<NavBar />);
    expect(container).toMatchSnapshot();
  });

  it('nav hides/show on Button toggle',async()=>{
    const { getByTestId } = render(<NavBar />);
    const nav=getByTestId('nav')
    expect(nav).not.toHaveClass('hide')
    const btns=getByTestId('button')
    fireEvent.click(btns);
    expect(nav).toHaveClass('hide')
  })

});
