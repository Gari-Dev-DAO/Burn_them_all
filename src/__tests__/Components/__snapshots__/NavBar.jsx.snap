// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders component 1`] = `
<div>
  <div
    class="navBar"
  >
    <div
      class="toggleBar"
    >
      <button
        data-testid="button"
        style="background-color: transparent; cursor: pointer; z-index: 200;"
      >
        <svg
          color="white"
          fill="currentColor"
          height="32"
          stroke="currentColor"
          stroke-width="0"
          style="color: white;"
          version="1.1"
          viewBox="0 0 17 17"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g />
          <path
            d="M17 2v1h-17v-1h17zM0 7h17v-1h-17v1zM0 11h17v-1h-17v1zM0 15h17v-1h-17v1z"
          />
        </svg>
      </button>
    </div>
    <div
      class="nav "
      data-testid="nav"
    >
      <div
        class="navLinkBox"
      >
        <a
          class="link"
        >
          Go bonk Errs
        </a>
      </div>
      <div
        class="navLinkBox navlinks"
      >
        <a
          class="link"
          href="https://www.coingecko.com/en/coins/bonk"
          target="_blank"
        >
          Buy Bonk
        </a>
        <a
          class="link"
          href="https://web.telegram.org/k/#-813184380"
          target="_blank"
        >
          Telegram
        </a>
      </div>
      <div
        class="navLinkBox"
      >
        <div>
          <button
            class="wallet-adapter-button walletBtn"
            tabindex="0"
            type="button"
          >
            Select Wallet
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
